skribenten-web/frontend/src/Brevredigering/LetterEditor/actions/toggleBulletList.ts

```diff
@@ -0,0 +1,436 @@
+  });
+
+  draft.redigertBrev.blocks[literalIndex.blockIndex] = newThisParagraph;
+  //---------------------------------------------
+
+  //---- Merging av punktlister dersom det finnes en punktliste-blokk før ----------------
+  const doesPreviousBlockEndWithAnItemList =
+    draft.redigertBrev.blocks[literalIndex.blockIndex - 1]?.content.at(-1)?.type === "ITEM_LIST";
+
+  const doesThisBlockStartWithAnItemList = newThisParagraph.content[0].type === "ITEM_LIST";
+
+  if (doesPreviousBlockEndWithAnItemList && doesThisBlockStartWithAnItemList) {
+    const previousBlock = draft.redigertBrev.blocks[literalIndex.blockIndex - 1];
+
+    const previousBlockContentBeforeItemList = previousBlock.content.slice(0, -1);
+    const previousBlockContentItemList = previousBlock.content.at(-1) as ItemList;
+
+    const thisBlockContentItemList = newThisParagraph.content[0] as ItemList;
+    const thisBlockContentAfterItemList = newThisParagraph.content.slice(1);
+
+    const thisBlockMergedWithPreviousBlock = newParagraph({
+      id: previousBlock.id,
+      content: [
+        ...previousBlockContentBeforeItemList,
+        newItemList({
+          items: [...previousBlockContentItemList.items, ...thisBlockContentItemList.items],
+          deletedItems: [...previousBlockContentItemList.deletedItems, ...thisBlockContentItemList.deletedItems],
+        }),
+        ...thisBlockContentAfterItemList,
+      ],
+      deletedContent: [...previousBlock.deletedContent, ...newThisParagraph.deletedContent],
+    });
+
+    const newBlocksWhereMergedWithPrevious = [
+      ...draft.redigertBrev.blocks.slice(0, literalIndex.blockIndex - 1),
+      thisBlockMergedWithPreviousBlock,
+      ...draft.redigertBrev.blocks.slice(literalIndex.blockIndex + 1),
+    ];
+
+    draft.redigertBrev.blocks = newBlocksWhereMergedWithPrevious;
+    draft.redigertBrev.deletedBlocks = [
+      ...draft.redigertBrev.deletedBlocks,
+      ...(newThisParagraph.id ? [newThisParagraph.id] : []),
+    ];
+  }
+  //---------------------------------------------
+
+  //her tar vi høyde for at det kan potensielt ha skjedd en merge eller ikke
+  const possibleNewThisBlockIndex = draft.redigertBrev.blocks.findIndex((block) =>
+    block.content.some(
+      (content) => content.type === "ITEM_LIST" && content.items.some((i) => isEqual(i.content, sentenceElements)),
+    ),
+  );
+  const possibleNewThisBlock = draft.redigertBrev.blocks[possibleNewThisBlockIndex];
+
+  const doesNextBlockStartWithAnItemList =
+    draft.redigertBrev.blocks[possibleNewThisBlockIndex + 1]?.content[0]?.type === "ITEM_LIST";
+  const doesThisBlockEndWithAnItemList = possibleNewThisBlock.content.at(-1)?.type === "ITEM_LIST";
+
+  //-- Merging av punktlister dersom det finnes en punktliste-blokk etter (merk at her bygger vi videre på det som potensielt er allerede blitt merget fra før)
+  if (doesThisBlockEndWithAnItemList && doesNextBlockStartWithAnItemList) {
```

Comment:
Hadde nok også vært nice om vi kunne fått til å generalisert koden slik at den kan gjenbrukes for merge med både forrige og neste blokk.